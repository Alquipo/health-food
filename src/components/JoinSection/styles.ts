import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.section`
  ${({ theme }) => css`
    display: flex;
    max-height: 70rem;

    ${media.lessThan('large')`
      padding: ${theme.spacings.xxlarge} ${theme.spacings.xsmall};
    `};
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    flex: 1;
    display: flex;
    align-items: center;
    justify-content: center;
    padding: ${theme.spacings.xsmall};

    ${media.lessThan('large')`
      align-items: center;
      justify-content: center;
    `};
  `}
`

export const FormWrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: 50rem;

    button {
      margin-left: ${theme.spacings.xxsmall};
    }
  `}
`

export const Title = styled.h1`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.medium};
    font-size: ${theme.font.sizes.huge};
    color: ${theme.colors.secondary};
    letter-spacing: 0.1rem;
    line-height: 1.3;

    ${media.lessThan('small')`
      font-size: ${theme.font.sizes.xxlarge};

    `};
  `}
`

export const Image = styled.img`
  flex: 1;
  object-fit: cover;

  ${media.lessThan('large')`
    display: none;
  `};
`
