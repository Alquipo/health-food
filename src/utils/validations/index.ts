import * as yup from 'yup'

const nascimentoRegex =
  /^((((19|[2-9]\d)\d{2})-(0[13578]|1[02])-(0[1-9]|[12]\d|3[01]))|(((19|[2-9]\d)\d{2})-(0[13456789]|1[012])-(0[1-9]|[12]\d|30))|(((19|[2-9]\d)\d{2})-02-(0[1-9]|1\d|2[0-8]))|(((1[6-9]|[2-9]\d)(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00))-02-29))$/g

const cpfRegex = /^\d{3}.?\d{3}.?\d{3}-?\d{2}$/

const cepRegex = /^([\d]{2})\.*([\d]{3})-*([\d]{3})/

const estadoRegex =
  /^(AC|AL|AP|AM|BA|CE|DF|ES|GO|MA|MT|MS|MG|PA|PB|PR|PE|PI|RJ|RN|RS|RO|RR|SC|SP|SE|TO)$/i

export const fieldsValidationSchema = yup.object({
  name: yup.string().required('Is not allowed to be empty'),
  birthDate: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(nascimentoRegex, 'must be a valid email'),
  cpf: yup
    .string()
    .max(14, 'Must be 14 characters')
    .required('Is not allowed to be empty')
    .matches(cpfRegex, 'Invalid CPF'),
  zipCode: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(cepRegex, 'Invalid ZipCode'),
  address: yup.string().required('Is not allowed to be empty'),
  district: yup.string().required('Is not allowed to be empty'),
  number: yup.number().required('Is not allowed to be empty'),
  city: yup.string().required('Is not allowed to be empty'),
  state: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(estadoRegex, 'Invalid State')
})
