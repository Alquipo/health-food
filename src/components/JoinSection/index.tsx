import * as S from './styles'

import { Formik, Form } from 'formik'

import Button from '../Button'
import TextField from '../TextField'

type FormType = {
  email: string
}

const initialValues: FormType = {
  email: ''
}

export default function JoinSection() {
  async function handleSubmit() {
    return null
  }

  return (
    <S.Wrapper>
      <S.Content>
        <S.FormWrapper>
          <S.Title>Join our membership to get special offer</S.Title>

          <Formik initialValues={initialValues} onSubmit={handleSubmit}>
            {() => (
              <Form style={{ display: 'flex' }}>
                <TextField
                  placeholder="Enter your email address"
                  name="email"
                  type="email"
                  fullWidth
                />
                <Button type="submit" aria-label="join">
                  Join
                </Button>
              </Form>
            )}
          </Formik>
        </S.FormWrapper>
      </S.Content>

      <S.Image src="/img/bloco_final_image.svg" />
    </S.Wrapper>
  )
}
