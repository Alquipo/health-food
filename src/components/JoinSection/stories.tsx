import { Story, Meta } from '@storybook/react'
import JoinSection from '.'

export default {
  title: 'JoinSection',
  component: JoinSection
} as Meta

export const Default: Story = () => <JoinSection />
