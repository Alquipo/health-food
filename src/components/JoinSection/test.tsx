import { fireEvent, render, screen, waitFor } from 'utils/test-utils'

import JoinSection from '.'

describe('<JoinSection/>', () => {
  it('should render a Join Section correctly', () => {
    const { container } = render(<JoinSection />)

    expect(screen.getByTestId(/email/i)).toBeInTheDocument()

    expect(screen.getByRole('button', { name: /Join/i })).toBeInTheDocument()

    expect(
      screen.getByRole('heading', {
        name: /Join our membership to get special offer/i
      })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should handle Join button ', () => {
    render(<JoinSection />)

    const handleSubmit = jest.fn()

    fireEvent.click(screen.getByLabelText(/join/i))

    waitFor(() => {
      expect(handleSubmit).toHaveBeenCalled()
    })
  })
})
