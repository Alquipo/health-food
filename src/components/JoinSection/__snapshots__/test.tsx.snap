// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JoinSection/> should render a Join Section correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-height: 70rem;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1.6rem;
}

.c2 {
  width: 100%;
  max-width: 50rem;
}

.c2 button {
  margin-left: 0.8rem;
}

.c3 {
  margin-bottom: 3.2rem;
  font-size: 4.8rem;
  color: #1D164D;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  line-height: 1.3;
}

.c7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  object-fit: cover;
}

.c6 {
  padding: 1.6rem 2.4rem;
  background: #BADC58;
  color: #FFFFFF;
  border-radius: 0.6rem;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
  cursor: pointer;
  border: none;
  font-size: 1.8rem;
  font-weight: 600;
}

.c6:hover {
  -webkit-filter: brightness(0.95);
  filter: brightness(0.95);
}

.c4 {
  width: 100%;
  max-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  grid-area: email;
}

.c5 {
  display: block;
  width: 100%;
  padding: 1.6rem;
  border: solid 0.1rem #1D164D;
  border-radius: 0.6rem;
  font-size: 1.6rem;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
  font-size: 1.4rem;
}

.c5:focus,
.c5:hover {
  outline: none;
}

@media (max-width:1170px) {
  .c0 {
    padding: 5.6rem 1.6rem;
  }
}

@media (max-width:1170px) {
  .c1 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:450px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (max-width:1170px) {
  .c7 {
    display: none;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h1
        class="c3"
      >
        Join our membership to get special offer
      </h1>
      <form
        action="#"
        style="display: flex;"
      >
        <div
          class="c4"
        >
          <input
            autocomplete="off"
            class="c5"
            data-testid="email"
            name="email"
            placeholder="Enter your email address"
            type="email"
            value=""
          />
        </div>
        <button
          aria-label="join"
          class="c6"
          type="submit"
        >
          Join
        </button>
      </form>
    </div>
  </div>
  <img
    class="c7"
    src="/img/bloco_final_image.svg"
  />
</section>
`;
